_ml-commons-plugin/conversational-search.md

```diff
@@ -0,0 +1,365 @@
+
+```json
+GET /_plugins/_ml/memory/conversation/4of2c9nhoIuhcr
+```
+
+The Memory API returns the following interaction information:
+
+```json
+{
+  "interactions": [
+    {
+      "interaction_id": "342968y2u4-0",
+      "conversation_id": "0y4hto_in1",
+      "create_time": "2023-4-23 10:25.324662",
+      "input": "How do I make an interaction?",
+      "prompt_template": "Hello OpenAI, can you answer this question? \
+											Here's some extra info that may help. \
+											[INFO] \n [QUESTION]",
+      "response": "Hello, this is OpenAI. Here is the answer to your question.",
+      "origin": "MyFirstOpenAIWrapper",
+      "additional_info": "Additional text related to the answer \
+											A JSON or other semi-structured response"
+	  }, ... (9 more since max_results defaults to 10)
+	],
+  "next_token": 10
+}
+```
+
+### Deleting conversations
+
+To delete a conversation, use the `DELETE` operation as showing in the following example:
+
+```json
+DELETE /_plugins/_ml/memory/conversation/4of2c9nhoIuhcr
+```
+
+The Memory API responds with the following:
+
+```json
+{ "success": true }
+```
+
+## RAG pipeline
+
+RAG is a technique for grounding LLMs and a natural language interface, implemented as a search pipeline. 
+
+### Enabling RAG
+
+Use the following cluster setting to enable the RAG pipeline feature:
+
+```json
+PUT /_cluster/settings
+{
+  "persistent": {"plugins.ml_commons.rag_pipline_feature_enabled": "true"}
+}
+```
+
+### Connecting the model
+
+RAG requires a LLM to function. We recommend using a [connector]({{site.url}}{{site.baseurl}}/ml-commons-plugin/extensibility/connectors/).
+
+In the following step through, we set up an HTTP connector using the OpenAI GPT 3.5 model. To set up that model:
+
+1. Use the Connector API to create the HTTP connector:
+
+  ```json
+  POST /_plugins/_ml/connectors/_create
+  {
+    "name": "OpenAI Chat Connector",
+    "description": "The connector to public OpenAI model service for GPT 3.5",
+    "version": 2,
+    "protocol": "http",
+    "parameters": {
+      "endpoint": "[api.openai.com](http://api.openai.com/)",
+      "model": "gpt-3.5-turbo",
+      "temperature": 0
+    },
+    "credential": {
+      "openAI_key": "<your OpenAI key>"
+   },
+    "actions": [
+      {
+        "action_type": "predict",
+        "method": "POST",
+        "url": "[https://$](https://%24/){parameters.endpoint}/v1/chat/completions",
+        "headers": {
+          "Authorization": "Bearer ${credential.openAI_key}"
+        },
+        "request_body": "{ \"model\": \"${parameters.model}\", \"messages\": ${parameters.messages}, \"temperature\": $  {parameters.temperature} }"
+      }
+    ]
+  }
+  ```
+
+2. Create a new model group for the connected model. You'll use the `model_group_id` returned by the to register the model:
```

Comment:
Complete this sentence:
You'll use the `model_group_id` returned by the _ _ _ _ to register the model:

Reply:
Fixed to "Register API."